sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldAccessDescriptor.java

```diff
@@ -316,11 +316,14 @@ public FieldAccessDescriptor withOrderByFieldInsertionOrder() {
     return toBuilder().setFieldInsertionOrder(true).build();
   }
 
-  /** Return the field ids accessed. Should not be called until after {@link #resolve} is called. */
-  public Set<Integer> fieldIdsAccessed() {
+  /**
+   * Return the field ids accessed. Should not be called until after {@link #resolve} is called.
+   * Iteration order is consistent with {@link #getFieldsAccessed}.
+   */
+  public List<Integer> fieldIdsAccessed() {
     return getFieldsAccessed().stream()
         .map(FieldDescriptor::getFieldId)
-        .collect(Collectors.toSet());
+        .collect(Collectors.toList());
   }
```

Comment:
maybe instead we should make getFieldsAccessed return a List, instead of storing it as a set internally?

Reply:
Yeah that's probably a good idea. I can add that change here.

Reply:
How does this look?